$OpenBSD: patch-xwindow_h,v 1.1 2013/05/24 16:15:24 dcoppa Exp $

commit 2d3103345e3a266eaf1d5261ef37e36ac5498ecc
Author: Uli Schlachter <psychon@znc.in>
Date:   Sun Apr 7 11:57:40 2013 +0200

Implement window gravity in ConfigureRequests

Previously, awesome didn't move windows when they sent a ConfigureRequest
(and it also shrunk them by the size of titlebars, before it forgot
to add these). This commit adds proper gravity support to
ConfigureRequest handling.

This was tested with test-gravity from metacity (in src/wm-tester).

Signed-off-by: Uli Schlachter <psychon@znc.in>

--- xwindow.h.orig	Mon Apr  1 13:44:46 2013
+++ xwindow.h	Fri May 24 16:58:00 2013
@@ -41,6 +41,7 @@ void xwindow_takefocus(xcb_window_t);
 void xwindow_set_cursor(xcb_window_t, xcb_cursor_t);
 void xwindow_set_border_color(xcb_window_t, color_t *);
 void xwindow_set_shape(xcb_window_t, int, int, enum xcb_shape_sk_t, cairo_surface_t *, int);
+void xwindow_translate_for_gravity(xcb_gravity_t, int16_t, int16_t, int16_t, int16_t, int16_t *, int16_t *);
 
 #endif
 // vim: filetype=c:expandtab:shiftwidth=4:tabstop=8:softtabstop=4:textwidth=80
